web/src/user_profile.ts

```diff
@@ -428,10 +507,13 @@ export function show_user_profile_access_error_modal() {
     });
 }
 
-export function show_user_profile(user, default_tab_key = "profile-tab") {
+export function show_user_profile(
+    user: CrossRealmBot | User,
```

Comment:
This can be changed to `User | {is_system_bot?: boolean}` to avoid all those extra key checkings but this felt more accurate.

Reply:
You meant `User & {is_system_bot?: boolean}` ? . Also the argument `user` in all the callers is of type `User` fetched from `get_by_user_id` from `people.ts`. This is leading to unconsistency in types, we should fix either one of the two function's type. Either this is only accepts type `User` or `get_by_user_id` returns `CrossRealmBot` also and needs refactor or type fix.

Reply:
Yeah, I meant `User & {is_system_bot?: boolean}`. Okay I'll check that.

Reply:
I think the return type of `get_user_id` and the type of `let people_by_user_id_dict: Map<number, User>` is incorrect. 

It should be `let people_by_user_id_dict: Map<number, User | CrossRealmBot>`

Reply:
That might warrant a prep commit fixing it.

Reply:
This seems like it can addressed in a follow-up.

Reply:
Well, except for the fact that this seems to be leading to wrong logic in https://github.com/zulip/zulip/pull/30755/commits/34e49d2718e92ef38b9c04abf66320c6ffdede29#r1699436444.

Reply:
Made a prep commit since it resolved all those key checks in the migration commit.